### PR TITLE
Add support to install a pre-commit hook

### PR DESCRIPTION
**Pull request by shan-96**
I always forget stuff like lint / test before a commit

- Add support to install a pre-commit git hook that will
    - check for Go and golangci-lint local versions and throw a warning if they are not desired versions
    - run lint check on the repo and fail if errors are found
    - run all tests and fail if errors are found
 - The hook can be installed by `make install`

### CODE REVIEW

File: scripts/pre-commit [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+./scripts/check_lint.sh
+
+echo "CGO_ENABLED = $CGO_ENABLED"
+
+if [ "$CGO_ENABLED" -eq "0" ] || [ -z "$CGO_ENABLED" ]
+then
+    make test-norace
+else
+    make test
+fi
```
Comment (ksysoev) [inline scripts/pre-commit:15]: @shan-96 could you clarify why do we check CGO here?

**Review by ksysoev — APPROVED** [review]
@shan-96 good idea, also time to time keep forgetting to run linter

**Comment by shan-96** [comment]
The flag `-race` only works when C executables for Go are enabled. Some
platforms might not even install while installing Go.

File: scripts/pre-commit (hunk 1 above)
Comment (shan-96) [reply scripts/pre-commit:15]: The go test flag `-race` only works when C executables for Go are enabled. Some platforms might not even install while installing Go.

Comment (ksysoev) [reply scripts/pre-commit:15]: @shan-96  I see, I didn't know..  on Mac it works without CGO starting from 1.20

> On macOS, the race detector has been rewritten not to use cgo: race-detector-enabled programs can be built and run without Xcode. On Linux and other Unix systems, and on Windows, a host C toolchain is required to use the race detector.

**Pull request merged**